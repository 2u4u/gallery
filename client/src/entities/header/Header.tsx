import React, { useRef, useState } from 'react';
import { useOutsideClick } from 'shared/hooks/useOutsideClick';
import { FileMimeType, View } from 'shared/types/index.type';
import HeaderSearch from './HeaderSearch';
import HeaderUpload from './HeaderUpload';
import HeaderView from './HeaderView';
import './index.scss';

type Props = {
    view: View,
    handleOpenModal: () => void;
    handleSearch: (e: React.ChangeEvent<HTMLInputElement>) => void;
    handleTypeSelect: (type?: FileMimeType) => void;
    changeView: (view: View) => void;
}

function Header({ handleOpenModal, handleTypeSelect, handleSearch, changeView, view }: Props) {
    const wrapperRef = useRef<HTMLInputElement>(null);
    const [openMenu, setOpenMenu] = useState(false);
    const handleToggleMenu = () => {
        setOpenMenu(openMenu => !openMenu);
    }
    const handleCloseMenu = () => {
        setOpenMenu(false);
    }
    
    useOutsideClick(wrapperRef, handleCloseMenu);

    return (
        <div className='header__block' ref={wrapperRef}>
            <button type='button' onClick={handleToggleMenu} className='header__mobile-button'>
                <img src="./img/menu.svg" alt="mobile menu" className=''/>
            </button>
            <div className={openMenu ? 'header__mobile-menu header__mobile-menu--open' : 'header__mobile-menu'}>
                <div className='header__actions'>
                    <HeaderView changeView={changeView} view={view} />
                    <HeaderUpload handleOpenModal={handleOpenModal} />
                </div>
                <HeaderSearch handleTypeSelect={handleTypeSelect} handleSearch={handleSearch} />
            </div>
        </div>
    );
}

export default Header;
