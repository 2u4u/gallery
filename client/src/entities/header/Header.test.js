import renderer from 'react-test-renderer';
import Header from './Header';
import { View } from 'shared/types/index.type';

describe('Header component ', () => {
    it('matches snapshot', () => {
        const view = View.grid;
        const handleOpenModal = () => false;
        const handleSearch = () => false;
        const handleTypeSelect = () => false;
        const changeView = () => false;
        const component = renderer.create(
            <Header 
                view={view}
                handleOpenModal={handleOpenModal}
                handleSearch={handleSearch}
                handleTypeSelect={handleTypeSelect}
                changeView={changeView}
            />,
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
