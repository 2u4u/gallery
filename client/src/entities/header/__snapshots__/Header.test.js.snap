// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component  matches snapshot 1`] = `
<div
  className="header__block"
>
  <button
    className="header__mobile-button"
    onClick={[Function]}
    type="button"
  >
    <img
      alt="mobile menu"
      className=""
      src="./img/menu.svg"
    />
  </button>
  <div
    className="header__mobile-menu"
  >
    <div
      className="header__actions"
    >
      <div
        className="items-view"
      >
        <button
          className="items-view__item"
          onClick={[Function]}
          type="button"
        >
          <img
            alt="list"
            src="./img/list.svg"
          />
        </button>
        <button
          className="items-view__item"
          onClick={[Function]}
          type="button"
        >
          <img
            alt="grid"
            src="./img/grid-active.svg"
          />
        </button>
      </div>
      <button
        className="header__upload upload"
        onClick={[Function]}
      >
        <img
          alt="upload file"
          className="upload__img"
          src="./img/upload.svg"
        />
        Upload file
      </button>
    </div>
    <div
      className="header__search search"
    >
      <input
        className="search__input"
        id="search"
        onChange={[Function]}
        placeholder="Search"
        type="text"
      />
      <label
        className="search__label"
        htmlFor="search"
      >
        <img
          alt="search"
          src="./img/search.svg"
        />
      </label>
      <div
        className="search__filter filter"
      >
        <div
          className="filter__block"
          onClick={[Function]}
        >
          <span
            className="filter__text"
          />
          <button
            className="filter__button"
            type="button"
          >
            <img
              alt="filter"
              src="./img/arrow-down.svg"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
